src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -796,6 +796,60 @@ mongoc_collection_count_with_opts (
    RETURN (ret);
 }
 
+/* --------------------------------------------------------------------------
+ *
+ * _make_aggregate_for_edc --
+ *
+ *       Construct an aggregate pipeline with the following form:
+ *
+ *
+ *       { pipeline: [
+ *           { $collStats: { count: {} } },
+ *           { $group: { _id: 1, n: { $sum: $count } } },
+ *         ]
+ *       }
+ *
+ *--------------------------------------------------------------------------
+ */
+static void
+_make_aggregate_for_edc (const mongoc_collection_t *coll, bson_t *out)
+{
+   bson_t pipeline;
+   bson_t coll_stats_stage;
+   bson_t coll_stats_stage_doc;
+   bson_t group_stage;
+   bson_t group_stage_doc;
+   bson_t sum;
+   bson_t cursor_empty;
+   bson_t empty;
+   const char *keys[] = {"0", "1"};
+   int key = 0;
+
+   bson_init (out);
+   BSON_APPEND_UTF8 (out, "aggregate", coll->collection);
+   BSON_APPEND_DOCUMENT_BEGIN (out, "cursor", &cursor_empty);
+   bson_append_document_end (out, &cursor_empty);
+   BSON_APPEND_ARRAY_BEGIN (out, "pipeline", &pipeline);
+
+   BSON_APPEND_DOCUMENT_BEGIN (&pipeline, keys[key], &coll_stats_stage);
+   key++;
```

Comment:
Nit: I feel like the `keys` variable doesn't bring any additional value:
```suggestion
   BSON_APPEND_DOCUMENT_BEGIN (&pipeline, "0", &coll_stats_stage);
```

Reply:
Done.